File: premium/web-frontend/modules/baserow_premium/automation/nodeTypes.js
import { LocalBaserowSignalTriggerType } from '@baserow/modules/automation/nodeTypes'
import { TriggerNodeTypeMixin } from '@baserow/modules/automation/nodeTypeMixins'
import { LocalBaserowRowCommentCreatedTriggerServiceType } from '@baserow_premium/integrations/localBaserow/serviceTypes'

export class LocalBaserowRowCommentCreatedTriggerNodeType extends TriggerNodeTypeMixin(
  LocalBaserowSignalTriggerType
) {
  static getType() {
    return 'local_baserow_row_comment_created'
  }

  getOrder() {
    return 5
  }

  get labelTemplateName() {
    return 'nodeType.localBaserowRowCommentCreatedLabel'
  }

  get dataType() {
    // The comment payload is a single object, unlike the row signals.
    return 'object'
  }

  get serviceType() {
    return this.app.$registry.get(
      'service',
      LocalBaserowRowCommentCreatedTriggerServiceType.getType()
    )
  }
}

File: premium/backend/src/baserow_premium/automation/nodes/node_types.py
from django.utils.translation import gettext_lazy as _

from baserow.contrib.automation.nodes.node_types import AutomationNodeTriggerType
from baserow_premium.automation.nodes.models import (
    LocalBaserowRowCommentCreatedTriggerNode,
)
from baserow_premium.integrations.local_baserow.service_types import (
    LocalBaserowRowCommentCreatedServiceType,
)


class LocalBaserowRowCommentCreatedNodeTriggerType(AutomationNodeTriggerType):
    type = "local_baserow_row_comment_created"
    model_class = LocalBaserowRowCommentCreatedTriggerNode
    service_type = LocalBaserowRowCommentCreatedServiceType.type
    display_name = _("Local Baserow row comment created")

    def get_pytest_params(self, pytest_data_fixture):
        service = pytest_data_fixture.create_local_baserow_row_comment_created_service()
        return {"service": service}
